packages/map/src/index.ts

```diff
@@ -127,7 +127,7 @@ export class ReactiveWeakMap<K extends object, V> extends WeakMap<K, V> {
   #keyTriggers = new TriggerCache<K>(WeakMap);
   #valueTriggers = new TriggerCache<K>(WeakMap);
 
-  constructor(initial?: [K, V][]) {
+  constructor(initial?: Iterable<readonly [K, V]> | null) {
```

Comment:
c.f. https://github.com/microsoft/TypeScript/blob/main/src/lib/es2015.iterable.d.ts#LL147C40-L147C67

Note that TypeScript doesn't support undefined or null; I think this is a bug because [the spec](https://tc39.es/ecma262/#sec-weakmap-iterable) says:

> 4. If iterable is either undefined or null, return map.

[The TS PR](https://github.com/microsoft/TypeScript/pull/43396) making "Map constructor argument optional and nullable" didn't include `WeakMap`, which I believe is an oversight. We'll see if/when I get enough nerve to submit a PR TypeScript 😅

Reply:
since we are doing a `if (initial)` check, we could allow any falsy type.
But we should probably limit to nullable if we decide to change the check